lib/SIL/InstructionUtils.cpp

```diff
@@ -246,9 +246,52 @@ void ConformanceCollector::scanType(Type type) {
         auto substs = SubType->gatherAllSubstitutions(M.getSwiftModule(),
                                                      nullptr);
         scanSubsts(substs);
+
+        // Check the types of all stored properties of the nominal type.
+        for (VarDecl *Field : NT->getStoredProperties()) {
+          if (Field->hasClangNode() || !Field->hasInterfaceType())
+            continue;
+          CanType FieldTy = SubType->getTypeOfMember(M.getSwiftModule(),
+                                          Field, nullptr)->getCanonicalType();
+          if (needsMetaData(FieldTy))
+            scanType(FieldTy);
+        }
+      }
+    }
+  });
+}
+
+bool ConformanceCollector::needsMetaData(Type type) {
+  type = type->getCanonicalType();
```

Comment:
Change this to SILType to remove unnecessary getCanonicalType() calls here and elsewhere